module/x/gravity/abci.go

```diff
@@ -247,35 +262,60 @@ func outgoingTxSlashing(ctx sdk.Context, k keeper.Keeper) {
 			if valInfo.exist && valInfo.sigs.StartHeight < int64(otx.GetCosmosHeight()) {
 				if _, ok := signatures[valInfo.val.GetOperator().String()]; !ok {
 					if !valInfo.val.IsJailed() {
+						power := valInfo.val.ConsensusPower(k.PowerReduction)
 						k.StakingKeeper.Slash(
 							ctx,
 							valInfo.cons,
 							ctx.BlockHeight(),
-							valInfo.val.ConsensusPower(k.PowerReduction),
+							power,
 							params.SlashFractionBatch,
 						)
 						k.StakingKeeper.Jail(ctx, valInfo.cons)
+
+						ctx.EventManager().EmitEvent(
+							sdk.NewEvent(
+								slashingtypes.EventTypeSlash,
+								sdk.NewAttribute(slashingtypes.AttributeKeyAddress, valInfo.cons.String()),
+								sdk.NewAttribute(slashingtypes.AttributeKeyJailed, valInfo.cons.String()),
+								sdk.NewAttribute(slashingtypes.AttributeKeyReason, types.AttributeMissingBridgeBatchSig),
```

Comment:
This code needs to be cleaned up. I think we have 1. slashing for missed sigs (not just batch sigs, all the sigs) 2. slashing for oracle stuff and I think theres a third one too. We could eliminate a bunch of params.

Reply:
Not sure I totally follow @jackzampolin. There are only two places where I saw that we jail and slash, which I've addressed in this PR.

I merged this PR. If there is more follow up needed, can you create an issue and I'll look into it pls?